### DISCUSSION TITLE
[Data-importer] Yunohost Firefly, self-hosted Data-importer, 500 error

### DISCUSSION THREAD

**Discussion by MokuJinJin** [firefly-iii/firefly-iii#6088] [Firefly III] [answered]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely me, not you](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

Whoops! 500 :(
Trying to import something to my newly install firefly with Yunohost
Just after oauth authorize, got this 500

### Expected behaviour

i don't know, next step to import

### Debug information

Debug information generated at 2022-05-22 01:22:27 Europe/Amsterdam for Firefly III Data Importer version **0.9.9**.

| Scope | Version |
| --- | --- |
| FIDI | 0.9.9 |
| Build | (unknown), (unknown) |
| PHP | 8.1.6 |
| Host | WINNT |

| System info | Value |
| --- | --- |
| Using docker? | false |
| System TZ | Europe/Amsterdam |
| Browser TZ | Europe/Paris |
| App environment | local |
| App debug mode | false |
| App cache driver | file |
| App logging | debug, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Interface | apache2handler |
| BCscale | 12 |
| Trusted proxies (.env) |  |

| User info | Value |
| --- | --- |
| User agent | Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:100.0) Gecko/20100101 Firefox/100.0 |

## Error message

JSON exception when decoding response: Syntax error 

## data-import-2022-05-22.log
```log
[2022-05-22 01:24:19] local.DEBUG: Will contact "https://yuno.mokujin.org/firefly/oauth/token" for a token.  
[2022-05-22 01:24:19] local.ERROR: JSON exception when decoding response: Syntax error  
[2022-05-22 01:24:19] local.ERROR: Response from server: "<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <title>YunoHost Portal</title>

  <!-- Responsive -->
  <meta name="format-detection" content="telephone=no" />
  <meta name="viewport" content="width=device-width, height=device-height, initial-scale=1" />

  <!-- Do not index SSOWat pages -->
  <meta name="robots" content="noindex, nofollow">

  <!-- Stylesheets -->
  <link rel="stylesheet" href="assets/css/ynh_portal.css">
  <link rel="stylesheet" href="assets/themes/default/custom_portal.css">

  <!-- Icons -->
  <link rel="shortcut icon" href="assets/icons/favicon.ico">
  <link rel="apple-touch-icon" sizes="57x57" href="assets/icons/apple-touch-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="114x114" href="assets/icons/apple-touch-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="72x72" href="assets/icons/apple-touch-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="144x144" href="assets/icons/apple-touch-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="60x60" href="assets/icons/apple-touch-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="120x120" href="assets/icons/apple-touch-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="76x76" href="assets/icons/apple-touch-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="152x152" href="assets/icons/apple-touch-icon-152x152.png">
  <link rel="icon" type="image/png" href="assets/icons/favicon-196x196.png" sizes="196x196">
  <link rel="icon" type="image/png" href="assets/icons/favicon-160x160.png" sizes="160x160">
  <link rel="icon" type="image/png" href="assets/icons/favicon-96x96.png" sizes="96x96">
  <link rel="icon" type="image/png" href="assets/icons/favicon-16x16.png" sizes="16x16">
  <link rel="icon" type="image/png" href="assets/icons/favicon-32x32.png" sizes="32x32">
  <meta name="msapplication-TileColor" content="#41444f">
  <meta name="msapplication-TileImage" content="/mstile-144x144.png">
</head>
<body class="ynh-user-portal ">

  <div id="ynh-logo" class="ynh-logo">
    <span class="element-invisible">Yunohost</span>
  </div>

  <div class="content">


    <div class="wrapper messages info">Please log in to access to this content</div>
<div class="ynh-wrapper login">
<form class="login-form" name="input" action="" method="post">
  <div class="form-group">
    <label class="icon icon-user" for="user"><span class="element-invisible">Username</span></label>
    <input id="user" type="text" name="user" placeholder="Username" class="form-text" autocomplete="username" autofocus required>
  </div>
  <div class="form-group">
    <label class="icon icon-lock" for="password"><span class="element-invisible">Password</span></label>
    <input id="password" type="password" name="password" placeholder="Password" class="form-text" autocomplete="current-password" required>
  </div>
  <input type="submit" value="Log in" class="btn classic-btn large-btn">
</form>
</div>

  </div>

  <!-- Scripts -->
  <script src="assets/js/ynh_portal.js"></script>
  <script src="assets/themes/default/custom_portal.js"></script>
</body>
</html>

"  
[2022-05-22 01:24:19] local.ERROR: JSON exception when decoding response: Syntax error {"exception":"[object] (App\\Exceptions\\ImporterErrorException(code: 0): JSON exception when decoding response: Syntax error at C:\\wamp64\\www\\data-importer\\app\\Http\\Controllers\\TokenController.php:115)
[stacktrace]
```

### Steps to reproduce

1. fresh install Firefly with Yunohost
2. create oauth client
3. install data-importer on another PC with wampp server
4. set VERIFY_TLS_SECURITY=false, i don't know why it doesn't recognize my official let's encrypt SSL
5. creating virtualhost with wampp server : http://firefly3-importer
6. accessing http://firefly3-importer/ adn giving Firefly III URL and client ID
7. accessing https://yuno.mokujin.org/firefly/oauth/ and authorise
8. http://firefly3-importer/callback Woops 500 error

### Additional info

i understand the importer can't read Json since it's Yunohost login form, but WHY can't it be auth ?

What i'm doing wrong ?

**Comment by JC5** [comment]
Your Yunohost has some kind of authentication page in front of Firefly III. So when the data importer sends you to Firefly III for an access token, you don't get the JSON content, you get Yunohost's authentication page. I can't fix that, I don't know how that works.

Maybe somebody else knows the solution?

**Reply by JesApp** [reply, marked as answer]
I "fixed" this by reinstalling the firefly-iii yunohost app and setting the "Should this app be exposed to anonymous visitors?" option to true.

This obviously isn't a real solution, since I really don't want to do that. I've started a discussion on this [here](https://forum.yunohost.org/t/firefly-iii-importer-can-not-connect-because-of-login-redirect/24047/2).
